test/util/testnode/read.go

```diff
@@ -39,6 +40,25 @@ func ReadBlockchain(ctx context.Context, rpcAddress string) ([]*types.Block, err
 	return ReadBlockHeights(ctx, rpcAddress, 1, status.SyncInfo.LatestBlockHeight)
 }
 
+// ReadBlockchainInfo retrieves the blockchain information from height 0 up to the latest height from the node at
+// rpcAddress and returns it.
+func ReadBlockchainInfo(ctx context.Context, rpcAddress string) (*ctypes.ResultBlockchainInfo, error) {
+	client, err := http.New(rpcAddress, "/websocket")
+	if err != nil {
+		return nil, err
+	}
+	resp, err := client.Status(ctx)
+	if err != nil {
+		return nil, err
+	}
+	lastHeight := resp.SyncInfo.LatestBlockHeight
+	res, err := client.BlockchainInfo(ctx, 0, lastHeight)
```

Comment:
This will on return up to 20 blocks so it will mean we miss data if the chain is longer than 20

Reply:
Mind elaborating why `20`? it reads from the first block up to the `lastHeight`, so the entire history of the chain

Reply:
Never mind, I see your point, the implementation limits the max height to 20, thanks, will work on a different method then.

Reply:
I refactored the code, and now we can read the entire chain's headers.